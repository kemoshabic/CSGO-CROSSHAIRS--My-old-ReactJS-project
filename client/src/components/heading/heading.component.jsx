import React from "react";

import "./heading.styles.scss";

export const Heading = () => {
  return (
    <div className="heading">
      <h1 className="heading__primary">CSGO: Crosshairs</h1>
      <h2 className="heading__secondary">
        The place to find your favourite professional CS:GO crosshairs
      </h2>
      <div className="heading__steps">
        <div className="heading__box">
          <div className="heading__num">
            <span>1</span>
          </div>
          <p className="heading__box--text">Sign In</p>
        </div>

        <div className="heading__box">
          <div className="heading__num">
            <span>2</span>
          </div>
          <p className="heading__box--text">Search</p>
        </div>

        <div className="heading__box heading__box--purple">
          <div className="heading__num ">
            <span className="heading__num--purple">3</span>
          </div>
          <p className="heading__box--text">Save settings & crosshairs</p>
        </div>
      </div>
    </div>
  );
};

export default Heading;
