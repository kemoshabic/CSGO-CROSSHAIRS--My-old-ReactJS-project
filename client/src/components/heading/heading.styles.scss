.heading {
  display: flex;
  flex-direction: column;
  align-items: center;

  &__primary {
    font-size: 5rem;
    color: #ffffff;
    font-weight: 400;
  }

  &__secondary {
    font-size: 3.2rem;
    font-weight: 400;
    padding-top: 1.2rem;
    background-image: linear-gradient(to right, #80ff72, #7ee8fa);
    background-clip: text;
    color: transparent;
    text-align: center;
  }

  &__steps {
    display: flex;
    padding-top: 3.2rem;
    @media only screen and (max-width: 25em) {
      // 400px
      flex-wrap: wrap;
      padding-bottom: 2rem;
      justify-content: center;
    }
  }

  &__box {
    background-color: #fd074e;
    padding: 0.5rem 2.5rem;
    padding-left: 4rem;
    margin-right: 2rem;
    border-radius: 2.5rem;

    position: relative;

    &--purple {
      background-color: #6843cf;
    }

    &--text {
      font-size: 1.6rem;
      font-weight: 400;
    }
    @media only screen and (max-width: 25em) {
      // 400px
      margin-bottom: 2rem;;
    }
  }
  &__num {
    position: absolute;
    left: 0;
    top: 3;
    background-color: #ffffff;
    width: 2.5rem;
    height: 2.5rem;
    border-radius: 50%;
    margin-left: 0.6rem;

    display: flex;
    justify-content: center;
    align-items: center;
    &--purple {
      color: #6843cf !important;
    }

    & span {
      color: #fd074e;
      font-size: 1.6rem;
    }
  }
}
