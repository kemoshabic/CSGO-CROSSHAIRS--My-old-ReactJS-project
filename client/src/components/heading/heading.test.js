import React from 'react';
import { shallow } from 'enzyme';
import { Heading } from './heading.component'

it('should render Heading component', () => {
  expect(shallow(<Heading />)).toMatchSnapshot();
});