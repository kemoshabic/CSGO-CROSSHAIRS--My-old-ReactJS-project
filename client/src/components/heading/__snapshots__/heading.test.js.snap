// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Heading component 1`] = `ShallowWrapper {}`;
